preload_apps.py

```diff
@@ -70,7 +70,7 @@ def get_args():
             target_json = json.load(target_json_file)
 
         registry_creds = []
-        if params.registry_creds_file:
+        if params.registry_creds_file and os.path.exists(params.registry_creds_file):
```

Comment:
open() raises an exception if a file does not exist or can't be opened. Wouldn't it be better to handle this exception?

Reply:
So, say, if the file exists and can't be opened, the script should fail?

Reply:
i think the better way to handle possible file not existing is to always just do:
```
try:
    with open(params.registry_creds_file) as f:
        ...
except FileNotFoundError:
    # This is okay
    pass
```
Then it handles @MrCry0's comment and is also more "fool-proof" in more generic settings where a file could disappear between checking for it existing and reading it.

Reply:
correct - and failing to *read* will give a different exception. the only foolproof way to try and read and see if a file doesn't exist is to catch an exception. I know it's an edge is this case - but it's just the canonical way to read things safely and catch the not-exists case.

Reply:
@mike-sul Now I see, your way is the best for such logic.

Reply:
>where a file could disappear between checking for it existing and reading it.

Again, if this file is defined and it magically disappear then it means an error situation, no point to continue since it will fails while trying to fetch images from 3rd party registries without auth.

Reply:
my point is this is generally how *not* to read a file correctly. my suggestion *always* works. This method works under this condition, so every time you do this type approach, you get people asking questions in a review and you have to explain yourself.

Reply:
Well, I've got different opinion on it. I don't believe in "general" solution, if this is the only correct way to read a file why it's not embedded into `open` call :). Quite often if a file is missing then it's an error/exception so that `try catch` is not needed at all.

Reply:
it just seems redundant. `exits` makes a `stat` to see if the file exists and `open` makes an `open` syscall and checks if the open failed because the file didn't exist.

but - you've argued for this passionately several times now. there's no point changing it.

Reply:
@mike-sul Would you mind adding your comment about the logic in the code? It helps a lot in the future.

Reply:
@MrCry0 I added the comment.

Reply:
Speaking of performance - https://therenegadecoder.com/code/how-to-check-if-a-file-exists-in-python/#performance.